client/src/plus/collections/frequently-viewed.tsx

```diff
@@ -109,47 +95,113 @@ function getNextFrequentlyViewedSerial(
   );
 }
 
-export function useFrequentlyViewed(limit: number, offset: number, setEnd) {
-  const [collection, setCollection] =
-    useState<FrequentlyViewedCollection | null>();
-  const [updated, setUpdated] = useState(false);
+export function useFrequentlyViewed(
+  limit: number = 0,
+  offset: number = 10,
+  setEnd?: (bool) => void
+): FrequentlyViewedCollection {
+  const [collection, setCollection] = useState<FrequentlyViewedCollection>({
+    article_count: 0,
+    created_at: new Date().toISOString(),
+    description: "Articles you viewed more than 2 times in the past 30 days.",
+    items: [],
+    name: "Frequently Viewed Articles",
+    updated_at: new Date().toISOString(),
+  });
 
   useEffect(() => {
     let freqViewed = getFrequentlyViewed();
     freqViewed = freqViewed.filter((val) => val.timestamps.length >= 2);
     if (limit + offset > freqViewed.length) {
-      setEnd(true);
+      setEnd && setEnd(true);
     }
+    let paged: FrequentlyViewedItem[] = freqViewed
+      .slice(0, limit + offset)
+      .map((val) => {
+        const lastModified = val.timestamps[0]
+          ? new Date(freqViewed[0].timestamps[0]).toISOString()
+          : new Date().toISOString();
```

Comment:
bit confused by the logic here - if the item has a most recent timestamp (which it always should) set lastmodified to the most recent timestamp of the first item in the whole collection? regardless, I believe we don't use `created_at` or `updated_at` anywhere in the item component, so these could be removed here and from the type.

Reply:
Definitely a hangover from the previous frequently viewed implementation. I hadn't realized the updated_at usage was removed from the collections :). Good one.